ramiro_serra/dif_driver_investigation.md

```diff
@@ -0,0 +1,53 @@
+# Differential wheeled robots
+
+A differential wheeled robot is a mobile robot whose movement is based on two separately driven wheels placed on either side of the robot body. It can thus change its direction by varying the relative rate of rotation of its wheels and hence does not require an additional steering motion.
+
+![Robot\](https://www.google.com/url?sa=i&rct=j&q=&esrc=s&source=images&cd=&ved=2ahUKEwjQgfuB67TkAhUbIrkGHcxLCWUQjRx6BAgBEAQ&url=http://planning.cs.uiuc.edu/node659.html&psig=AOvVaw2lkNnIE3pIZqVMvjR70tXw&ust=1567606381458775 )](http://planning.cs.uiuc.edu/img5518.gif)
+
+###### Differential wheeled robot
+
+### How it works?
+
+If both the wheels are driven in the same direction and speed, the robot will go in a straight line. If both wheels are turned with equal speed in opposite directions, as is clear from the diagram shown, the robot will rotate about the central point of the axis. Otherwise, depending on the speed of rotation and its direction, the center of rotation may fall anywhere on the line defined by the two contact points of the tires. While the robot is traveling in a straight line, the center of rotation is an infinite distance from the robot. Since the direction of the robot is dependent on the rate and direction of rotation of the two driven wheels, these quantities should be sensed and controlled precisely. 
+
+To construct a simple model of the constraints that arise from the differential drive, only the distance ![ L](http://planning.cs.uiuc.edu/img83.gif) between the two wheels, and the wheel radius, ![ r](http://planning.cs.uiuc.edu/img165.gif), are necessary. The action vector ![ u = (u_r,u_l)](http://planning.cs.uiuc.edu/img5520.gif) directly specifies the two angular wheel velocities (e.g., in radians per second). Consider how the robot moves as different actions are applied.. If ![ u_l = u_r > 0](http://planning.cs.uiuc.edu/img5521.gif), then the robot moves forward in the direction that the wheels are pointing. The speed is proportional to ![ r](http://planning.cs.uiuc.edu/img165.gif). In general, if ![ u_l = u_r](http://planning.cs.uiuc.edu/img5522.gif), then the distance traveled over a duration ![t](http://planning.cs.uiuc.edu/img1142.gif) of time is ![ r t u_l](http://planning.cs.uiuc.edu/img5523.gif) (because ![ t u_l](http://planning.cs.uiuc.edu/img5524.gif) is the total angular displacement of the wheels). If ![u_l = -u_r \not = 0](http://planning.cs.uiuc.edu/img5525.gif), then the robot rotates clockwise because the wheels are turning in opposite directions. This motivates the placement of the body-frame origin at the center of the axle between the wheels. By this assignment, no translation occurs if the wheels rotate at the same rate but in opposite directions.
```

Comment:
Could you look in Gazebo which link acts as the body-frame origin as you mentioned in the text?